src/Cli/dotnet/commands/dotnet-test/Program.cs

```diff
@@ -118,15 +118,21 @@ private static TestCommand FromParseResult(ParseResult result, string[] settings
                 "-nologo"
             };
 
-            msbuildArgs.AddRange(result.OptionValuesToBeForwarded(TestCommandParser.GetCommand()));
+
+            var parsedArgs =
+                    result.OptionValuesToBeForwarded(TestCommandParser.GetCommand()) // all msbuild-recognized tokens
+                    .Concat(result.UnmatchedTokens); // all tokens that the test-parser doesn't explicitly track
```

Comment:
this is the real fix - this method doesn't call `ParseResult.GetArguments()` and so it only receives the 'understood' options/arguments on the command line. Since properties aren't added to the grammar for `dotnet test` (yet), the unknown tokens that they end up in should be forwarded along to MSBuild unchanged.